### ISSUE TITLE
Segfault on start

### ISSUE THREAD

**Issue by wefefgergergwgwg** [Facepunch/garrysmod-issues#1833]
https://bpaste.net/show/1f9e42e2058d

I dunno why does it fail, but it fails on debugger when I remember password in Steam.

**Comment by Bo98** [comment]
What graphics drivers are you using?

**Comment by d10sfan** [comment]
Getting same issue on Xubuntu 14.10. Update before this worked fine

**Comment by lizer261** [comment]
Getting same issue on Linux 14 server. Before update all was fine.

**Comment by lizer261** [comment]
What's new?

**Comment by Bo98** [comment]
It's been reproduced by the dev team and is being looked into I believe.

**Comment by lizer261** [comment]
Ok, the easy way:  install windows, lol.

**Comment by JOE95443** [comment]
I'm also having this problem on Ubuntu 14.04 32bit and Ubuntu 14.10 64bit. It seems to happen whenever a second person joins the server, or when the map changes automatically (TTT gamemode). I can't reproduce the problem in Virtualbox running the same OS.

**Comment by jgkamat** [comment]
+1

**Comment by lexxus-lex** [comment]
ok... 
3 days have passed, still the developers didn't change anything.

all on linux have this bug.

**Comment by d10sfan** [comment]
Would also like to have more information, since the game is completely unplayable.

**Comment by xalalau** [comment]
Any progress?? The game doesn't even show the menu...
I'm using Ubuntu and Linux Mint (both 64 bits) with an Intel i5 (Sandy Brigde).

**Comment by SeamusFD** [comment]
So it would appear the only people who can play garrys mod are windows users... ok then

**Comment by KintaroBC** [comment]
Exact same issue as everyone is having here on Fedora 21.

I enabled console logging and these error occur before it crashes:
Error: Material "debug/debuglightmap" uses unknown shader "LightmappedGeneric_dx8"
Error: Material "debug/debugluxels" uses unknown shader "DebugLuxels"

Though, I doubt they're related.

**Comment by JOE95443** [comment]
I get the same shader errors on Ubuntu, but my server doesn't crash until a second person joins the server, or when the map changes. I only get that error during startup.

**Comment by lexxus-lex** [comment]
For developers:

 Fontconfig error: "/etc/fonts/conf.d/10-scale-bitmap-fonts.conf", line 70: non-double matrix element
 Fontconfig error: "/etc/fonts/conf.d/10-scale-bitmap-fonts.conf", line 70: non-double matrix element
 Fontconfig warning: "/etc/fonts/conf.d/10-scale-bitmap-fonts.conf", line 78: saw unknown, expected number
Fontconfig warning: "~/.fonts.conf", line 1035: saw number, expected matrix
/home/lexxus/.local/share/Steam/SteamApps/common/GarrysMod/hl2.sh: строка 67:  4692 Ошибка сегментирования                   (подготовлен дамп ядра) ${GAME_DEBUGGER} "${GAMEROOT}"/${GAMEEXE} "$@"
Game removed: AppID 4000 "Garry's Mod", ProcID 4692 
[2015-03-12 15:04:47] Startup - updater built Feb 18 2015 15:31:44
Looks like steam didn't shutdown cleanly, scheduling immediate update check
</code>

**Comment by wefefgergergwgwg** [comment]
I am using nvidia blob with bumblebee (primus backend), if you want to know.

>Ok, the easy way: install windows, lol.
No.

**Comment by JOE95443** [comment]
Deleting the `/garrysmod/sv.db` file fixed the problem for me. Since deleting it my server hasn't crashed (been running with 10+ for almost an hour now). Before deleting the file, it would crash each time someone joined or when the map changed.

This fix worked on CentOS 7 64bit and Ubuntu 14.10 64bit.

Not sure if this is a permanent fix or not, but it's a step in the right direction.

**Comment by robotboy655** [comment]
What that does is empties the SQLLite database your addons might have been using to store data.

**Comment by JOE95443** [comment]
It's crashed 5 times since I posted that, so it's still a crippled piece of software. Think I'll just find another game to host.

**Comment by lexxus-lex** [comment]
<b>JOE95443</b>, i don't have sv.db

**Comment by KintaroBC** [comment]
This thread is about a startup segfault for the client, not the server. It's been a week, how come nobody from Facepunch has chimed in on this?

**Comment by xalalau** [comment]
I think they want we start to go pirate.

**Comment by wefefgergergwgwg** [comment]
They just do not care, they have Rust.

**Comment by TheFreeman193** [comment]
Many of the issues are being worked on currently. This was a big update, so inevitably there are a lot of problems to be resolved.

**Comment by shaunbc** [comment]
Is there anyone to fallback to prior to the update?

**Comment by xalalau** [comment]
Ty, The Freeman 193. But damn... If this has had happened in Windows the fix would be done in the next day. I hope that these bugs doesn't stay around forever like the awesomium issue with the checkboxes (I really fucking hate this one!! It's the only thing that ruins the user GMod experience on Linux). 

Well, meanwhile there is a video and an addon here waiting to be continued... Delayed for a week now. So, please, Gaben! Solve this demand problem with SteamOS, Vulkan and everything else!!

Anyway, ty for replying etc. GMod S2

**Comment by shaunbc** [comment]
I found a workaround. Just tested it on KDE Mint Linux 17 64-bit and it works flawlessly:
1. Install PlayOnLinux through your software manager.
2. Install Steam through PlayOnLinux, following ALL instructions.
3. Login to Steam through PlayOnLinux Run command.
4. Download and install Garry's Mod like you would regularly on the Linux version.
5. I tried running the game on all graphic settings cut either off or low so you could try and see if it works for you too.

**Comment by lexxus-lex** [comment]
<b>shaunbc</b>, this does not solve the problem on Linux :)

A quick solution: Install Windows :))))

**Comment by KintaroBC** [comment]
Buy us Windows, my email is: john@johntate.org

**Comment by shaunbc** [comment]
![gmod_shots](https://cloud.githubusercontent.com/assets/11476000/6655998/0d188320-caef-11e4-8045-87112347d8c5.png)

Yes, it DOES work. :+1:

**Comment by Random0824** [comment]
@shaunbc 
that's not the point, we're trying to fix it for linux not have to use play on linux.

**Comment by Random0824** [comment]
@Shaunbc whenever i try to press any key on playonlinux gmod, it crashes.

**Comment by PotcFdk** [comment]
@Random0824 This ticket is not about bugs related to Garry's Mod on PlayOnLinux. Please move that elsewhere.

**Comment by xalalau** [comment]
Shaunbc, PlayOnLinux is just Wine and Wine is all about Windows stuff. We are Linux, we want the Linux version working natively - without FPS drops and even more random crashes. Ty, I'll this "workaround".

**Comment by jgkamat** [comment]
@lizer261 @lexxus-lex +1 I want a windows copy too! Could you please fix all the random windows crashes too while you're giving me that license as well? Maybe then we can all play garrys mod with no OS and no actual application crashes!
My email is github@jgkamat.33mail.com

**Comment by thegrb93** [comment]
F off, linux hippie freeloaders.

**Comment by d10sfan** [comment]
@thegrb93 Tell me how buying the game and excepting it to work is considered "freeloading"

**Comment by SilverWingedSeraph** [comment]
@thegrb93 We bought the game, we want it to work. I've supported GMod since 9, and have recommended it to countless friends, who have since bought it. How am I a freeloader?

**Comment by thegrb93** [comment]
@d10sfan You're asking for software, which is not free, for free. How is that not freeloading?

**Comment by NothingMuchHereToSay** [comment]
@thegrb93 wut

You make no sense, we're asking an owner of a video game that's proprietary and closed source to fix the damn game. Imagine you buying Assassins Creed: Unity or which ever game it was that Ubisoft screwed up big time. Care to fix it yourself? No? Oh okay, too bad, you bought the game anyways and we're not going to fix it.

Stick to what you "like" or as I'd like so phrase it "what you're used to" and enjoy getting spied on unknowingly by Microsoft or Apple, whichever company's products you enjoy using.

And I like how these people are getting pissy over something that doesn't involve them at all. Either play your games or do your research that way you'd understand the differences between the "free" as you're using it and "freedom" as most of us use it.

Game devs, do you need someone from the FOSS community to ask a particular developer of a piece of the game to do something else again because of the lack of communication? I'll be happy to oblige.

Oh wait, this bug wasn't fixed either, https://github.com/Facepunch/garrysmod-issues/issues/86

Could this and that bug be related at all?

**Comment by thegrb93** [comment]
@NothingMuchHereToSay I'm talking about the people demanding to be given a free copy of windows. You missed the context of the situation drastically.

**Comment by yawkat** [comment]
@thegrb93 Pretty sure they were being sarcastic.

**Comment by thegrb93** [comment]
>Buy us Windows, my email is: john@johntate.org
Yeah, no.

**Comment by thegrb93** [comment]
Anyway, the Windows version is pretty messed up too so you're better off playing something else until this update gets fixed.

**Comment by KintaroBC** [comment]
Stop feeding the trolls guys.

**Comment by jgkamat** [comment]
@KintaroBC: +1 
This is an issue on the linux version of garrys mod, if you do not have debug info or something else to help the devs fix this issue (segfaulting on linux), please make a separate issue (or comment on a separate website) with your problems. Being a dev of a large project is hard as it is, spamming them with useless notifications will hurt us when one of us actually has useful information to give to them.

**Comment by Random0824** [comment]
@NothingMuchHereToSay 
That was fixed by another user, never implemented into the game but it was still a common issue and had a fix for it, the game still works with that bug.
For whatever this bug is, the game doesnt start past the blue screen.

**Comment by travka** [comment]
any progress ?

**Comment by lexxus-lex** [comment]
nope :(

developers don't know how fix own product

**Comment by philxyz** [comment]
This one and #1895 were both closed as duplicates of each other. It looks like Linux client is not supported any more. Especially when Lua tweaks come first. Please don't make me install Windows. Please?...

**Comment by TheFreeman193** [comment]
@philxyz 1834, 1838, 1869, 1895 were all closed as duplicates of this ticket, which remains open.

**Comment by d10sfan** [comment]
This is basically false advertising on the steam store now, there's a Linux icon but no one can play. This company has a similar track record with Rust, although at least they have some excuse with that being early access. They should have tested the game on Linux before releasing a patch that broke it for everyone.

Plus two weeks without any news or updates shows how little these developers care about paying customers.

**Comment by philxyz** [comment]
I would understand if they said Gmod 2 will be platform X or Y only, but in this case, the game has been taken away.

**Comment by TheFreeman193** [comment]
Linux support has not been removed. There are currently some complex issues on Linux and OSX involving the new font renderer, which will eventually be resolved. The changes were available on the dev branch for some time, and had the scale of the issues been apparent, the update would not have been released. The devs haven't forgotten, given up, or stopped caring. I understand it's frustrating for a lot of users and server owners, but we just need to hang in there, guys. :+1:

**Comment by philxyz** [comment]
Well, that's fine - as long as we know that someone is dealing with it.

**Comment by xalalau** [comment]
@TheFreeman193 I've been reading this post every day hoping for a "We're doing something.". Thank U.

**Comment by kurisu28** [comment]
My error looks a little different, but it crashes roughly 2 seconds into the "Loading..." screen before I even get to the main menu.
http://pastebin.ubuntu.com/10650549/
My system is an i7-4790K, GTX 760, 12 GB of RAM running Mint 17 with proprietary Nvidia drivers version 346.47, X server version 1.15.1 and Linux 3.13.0-24, for what that's worth.

**Comment by SilverWingedSeraph** [comment]
@kurisu28 If your issue is different, open a different issue. Upper right corner.

**Comment by c1yd3i** [comment]
I'm on Arch Linux *(x86_64 Linux 3.19.2-1-ARCH)*. 

Garry's Mod worked 100% fine before the update but now the game only gets to the loading screen and it immediately crashes. I'm not sure why it can't find the client libraries, they're there so I'm assuming it's an unrelated issue. Issues with font renderers were addressed above, I'm using [Infinality](https://wiki.archlinux.org/index.php/Infinality). 

Output of ```./hl2.sh ```:
```
SDL video target is 'x11'
SDL video target is 'x11'
SDL failed to create GL compatibility profile (whichProfile=0)!
Setting breakpad minidump AppID = 4000
dlopen failed trying to load:
/home/c1yd3i/.local/share/Steam/linux32/libsteam.so
with error:
/home/c1yd3i/.local/share/Steam/linux32/libsteam.so: cannot open shared object file: No such file or directory
dlopen failed trying to load:
libsteam.so
with error:
libsteam.so: cannot open shared object file: No such file or directory
Steam_SetMinidumpSteamID:  Caching Steam ID:  76561198058648898 [API loaded no]
Did not detect any valid joysticks.
Could not load program cache file glbaseshaders.cfg
Could not find base GL shader cache file
Could not load program cache file glshaders.cfg
Could not find user GL shader cache file
Precache: Took 0 ms, Vertex 0, Pixel 0
Couldn't load: \home\c1yd3i\.local\share\Steam\steamapps\common\GarrysMod\garrysmod\bin\client.so
Reason: \home\c1yd3i\.local\share\Steam\steamapps\common\GarrysMod\garrysmod\bin\client.so: cannot open shared object file: No such file or directory
```

**Comment by alien2003** [comment]
pls fix it or give my back my money or send me a free windows copy + money for new HDD (i don't trust Windows and I don't want it to send my data to Micro$oft or to NSA so I would like to have that shit on the separate drive): alien@openmailbox.org

**Comment by tkriik** [comment]
Running `ldd` with a few filtering commands on all the shared libraries under `~/.local/share/Steam/steamapps/common/GarrysMod/bin` shows that the following libraries are not found:

* libbass.so 
* libcairo.so.2 
* libcef.so 
* libgdk_pixbuf-2.0.so.0 
* libgdk-x11-2.0.so.0 
* libgtk-x11-2.0.so.0 
* libnspr4.so 
* libnss3.so 
* libnssutil3.so 
* libpango-1.0.so.0 
* libpangocairo-1.0.so.0 
* libplc4.so 
* libSDL2-2.0.so.0 
* libsmime3.so 
* libsteam_api.so 
* libtier0.so 
* libtogl.so 
* libvstdlib.so

I don't have the previous version at hand so I can't compare older results. I know that pango and cairo are used for font rendering though. Could this be related to the issue?

**Comment by travka** [comment]
any progress ? 2 weeks without results

**Comment by auggy** [comment]
@travka there was a progress report about 7 or 8 replies ago.

**Comment by Bo98** [comment]
Test the dev branch.

**Comment by travka** [comment]
pid 16646 != 16645, skipping destruction (fork without exec?)
ERROR: ld.so: object '/home/tr4ka/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/tr4ka/.local/share/Steam/ubuntu12_64/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS64): ignored.
SDL video target is 'x11'
SDL video target is 'x11'
SDL failed to create GL compatibility profile (whichProfile=0)!
Installing breakpad exception handler for appid(gameoverlayui)/version(20150218153205)
Installing breakpad exception handler for appid(gameoverlayui)/version(1.0)
Installing breakpad exception handler for appid(gameoverlayui)/version(1.0)
Setting breakpad minidump AppID = 4000
Steam_SetMinidumpSteamID:  Caching Steam ID:  76561198046448492 [API loaded no]
Installing breakpad exception handler for appid(gameoverlayui)/version(1.0)
Did not detect any valid joysticks.
Could not load program cache file glbaseshaders.cfg
Could not find base GL shader cache file
Could not load program cache file glshaders.cfg
Could not find user GL shader cache file
Precache: Took 0 ms, Vertex 0, Pixel 0
ConVarRef mat_dxlevel doesn't point to an existing ConVar
Game.so loaded for "Garry's Mod"
/home/tr4ka/.local/share/Steam/steamapps/common/GarrysMod/hl2.sh: line 67: 16648 Segmentation fault      (core dumped) ${GAME_DEBUGGER} "${GAMEROOT}"/${GAMEEXE} "$@"
Game removed: AppID 4000 "Garry's Mod", ProcID 16648

**Comment by Bo98** [comment]
Looks like the Linux version of the dev branch hasn't updated yet. I'll let you know when it does.

**Comment by d10sfan** [comment]
I haven't seen any official response here from the devs.

**Comment by c1yd3i** [comment]
How hard can it be to rollback the Linux updates? I mean, at least in the interim, so, you know, we can actually play the game... Doesn't seem like they care very much.

**Comment by robotboy655** [comment]
Should be fixed on the Dev branch, please test.

**Comment by KintaroBC** [comment]
The development branch isn't working for me, I'm running Fedora 21 with the rpmfusion nvidia drivers.

**Comment by travka** [comment]
Comfirmed, problem solved.

**Comment by robotboy655** [comment]
@KintaroBC, The update just went live ( 5 mins ago ), try again.

**Comment by cpk91898** [comment]
For me too, on Ubuntu 14.04  dev branch working.

**Comment by KintaroBC** [comment]
It's working, horray!

**Comment by dsneddon** [comment]
It works for me now on Fedora 20. Note that the dev branch did not automatically update, and I had to wait about 15 minutes after robotboy655's announcement to do the following:
1) Changed Garry's Mod branch to regular non-beta and downloaded update
2) Changed Garry's Mod back to dev branch and downloaded update.

After that, GMod works perfectly in F20, and so fast, even on my Intel i3000 integrated chipset. Great work, Facepunch, thanks for resolving our issues.

**Comment by tkriik** [comment]
Thanks for fixing the issue. One thing though—all the in-game menu fonts are in italic. What could be causing this?

**Comment by Bo98** [comment]
The new font renderer was reverted.

**Comment by robotboy655** [comment]
Yeah, fonts are back to being shit.

**Comment by lexxus-lex** [comment]
omg, they did it. Problem Solved

**Comment by xalalau** [comment]
"Solved"... I wanted these good looking fonts =(

**Comment by SilverWingedSeraph** [comment]
Can we close this now?

**Comment by Bo98** [comment]
It is.

**Comment by Slush97** [comment]
how do you download the update, its still not working for me

**Comment by jgkamat** [comment]
go to Garry's mod options, beta's and select the dev beta opt in.

**Comment by robotboy655** [comment]
@jgkamat, No.

Game updates are downloaded by Steam automatically unless you disabled automatic updates.

**Comment by ParkerCraftLite** [comment]
I'm running into this error still, even after reinstalling

Xubuntu 14.10 64-bit
FGLRX 14.501.1003, AMD R9 270

**Comment by Bo98** [comment]
Using AMD Catalyst drivers?

**Comment by ParkerCraftLite** [comment]
Yes

**Comment by Bo98** [comment]
I hear AMD Catalyst 15.3 beta fixes the issue but I don't know how widely available it is at the moment.

**Comment by ParkerCraftLite** [comment]
The latest available Linux Beta is 14.6, while the latest stable is 14.12. I don't think they update the public beta often.
